web/server/entrypoint.go

```diff
@@ -280,6 +283,9 @@ func (s *robotServer) serveWeb(ctx context.Context, cfg *config.Config) (err err
 				case <-doneServing:
```

Comment:
While we're here, I'd recommend removing this whole `case <-doneServing` stuff (and incidentally the select statement) and move straight to the killing/logging.